src/reducers/Material.js

```diff
@@ -43,6 +43,7 @@ function materialsToggleIsNonPeriodicForOne(state, action) {
         return state;
     }
     newMaterial.isNonPeriodic = !newMaterial.isNonPeriodic;
+    newMaterial.toCartesian()
```

Comment:
We need the units of the basis in cartesians for the non-periodic system. Otherwise, the units are in terms of `crystal` and get retransformed to cartesian every time the toggle switch, which causes the lattice size to change with every toggle.

Example
initial crystal coordinates
Si 0.0, 0.0, 0.0
Si 0.25, 0.25, 0.25

Crystal --> Cartesian transformation on first click of toggle isNonPeriodic switch
Si 0.0 0.0 0.0
Si 0.967, 0.967, 0.967
distance between coordinates = 0.967 angstrom between coordinates

Cartesian --> Crystal transformation as basis is saved for new material after lattice and basis are updated.
Si 0.375, 0.375, 0.375
Si 0.625, 0.625, 0.625

Crystal --> Cartesian  transformation on second click of toggle isNonPeriodic switch
Si 1.256, 1.265, 1.256
Si 2.093, 2.093, 2.093
distance between coordinates = 0.837 Angstrom between coordinates

The conversion from crystal to cartesian alters the points in a way that changes the distance between them, causing the lattice to shrink.

Reply:
> The conversion from crystal to cartesian alters the points in a way that changes the distance between them, causing the lattice to shrink.

It means the function that does the calculation is wrong then, doesn't it?